toolkit/tools/pkgworker/pkgworker.go

```diff
@@ -299,7 +299,7 @@ func tdnfInstall(packages []string) (err error) {
 		return
 	}
 
-	installArgs := []string{"install", "-y", releaseverCliArg}
+	installArgs := []string{"install", "-y", releaseverCliArg, "--disablerepo=mariner-official-base"}
```

Comment:
This is one change that I am slightly skeptical of, though it doesn't seem to provide a negative impact to the build process. I added this since the worker chroot has not internet connection and therefore can't access the base repo anyway. By doing so, this allows us to narrow down errors a bit better than simply many cURL failures that can mask other problems within the tdnf install when building packages.

Reply:
I share your skepticism.:) The main reason is that we should not hard-code anything external to the toolkit inside its code. The toolkit should only need to know about the internal RPM repos present in the chroot like the ones defined in `toolkit/resources/manifests/package/local.repo`. The `mariner-official-base` repo is not really part of the toolkit, may change at any time, and in general the toolkit should not have any dependency on such external entities. We do depend on the repo being in the chroot for the sake of fetching packages, but the dependency is on the chroot packages providing the repo configurations, not on the actual repo names.

Also, I wouldn't worry about the worker accidentally reaching out to any unwanted repositories. If I remember correctly, this code is run when we already downloaded/built all of the required packages, so they are available in the local RPM repos. If we see it reaching out elsewhere, that would probably (I might have forgotten some use case) be a bug, which we should address separately, because it would mean we have a more sever design flaw somewhere in our code. Your current change might hide that potential sever issue, so I'd prefer to revert it.

In other words: it's good to make sure we don't try to access the wrong repos, but I wouldn't include such "just in case" updates in this PR as that's our-of-scope and if there really is an issue for us to address here, it might require more investigation and a fix elsewhere in our design.